kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/RestClient.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.kyuubi.client.exception.KyuubiRestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestClient implements AutoCloseable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RestClient.class);
+
+  private CloseableHttpClient httpclient;
+
+  private String baseUrl;
+
+  private String authHeader;
+
+  public RestClient(String baseUrl, CloseableHttpClient httpclient, String authHeader) {
+    this.httpclient = httpclient;
+    this.baseUrl = baseUrl;
+    this.authHeader = authHeader;
+  }
+
+  @Override
+  public void close() throws Exception {
+    if (httpclient != null) {
+      httpclient.close();
+    }
+  }
+
+  public <T> T get(String path, Map<String, Object> params, TypeReference<T> type)
+      throws KyuubiRestException {
+    try {
+      String responseJson = get(path, params);
+      return new ObjectMapper().readValue(responseJson, type);
+    } catch (JsonProcessingException e) {
+      throw new KyuubiRestException(
+          "cannot convert response json to object: " + type.getType().getTypeName(), e);
+    }
+  }
+
+  public String get(String path, Map<String, Object> params) throws KyuubiRestException {
+    return doRequest(buildURI(path, params), RequestBuilder.get());
+  }
+
+  public <T> T post(String body, TypeReference<T> type) throws KyuubiRestException {
+    try {
+      String responseJson = post(body);
+      return new ObjectMapper().readValue(responseJson, type);
+    } catch (JsonProcessingException e) {
+      throw new KyuubiRestException(
+          "cannot convert response json to object: " + type.getType().getTypeName(), e);
+    }
+  }
+
+  public String post(String body) throws KyuubiRestException {
+    RequestBuilder postRequestBuilder =
+        RequestBuilder.post().setEntity(new StringEntity(body, StandardCharsets.UTF_8));
+    return doRequest(buildURI(), postRequestBuilder);
+  }
+
+  public String delete(String path, Map<String, Object> params) throws KyuubiRestException {
+    return doRequest(buildURI(path, params), RequestBuilder.delete());
+  }
+
+  private String doRequest(URI uri, RequestBuilder requestBuilder) throws KyuubiRestException {
+    String response = "";
+    CloseableHttpResponse httpResponse = null;
+    try {
+      HttpUriRequest httpRequest =
+          requestBuilder
+              .setUri(uri)
+              .setHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+              .setHeader(HttpHeaders.AUTHORIZATION, authHeader)
+              .build();
+
+      LOG.info("Executing {} request: {}", httpRequest.getMethod(), uri);
+      httpResponse = httpclient.execute(httpRequest);
```

Comment:
what happens if the returned status code is not 200

Reply:
The logic is in `BaseResponseHandler`:
```
httpResponse = httpclient.execute(httpRequest);
response = new BasicResponseHandler().handleResponse(httpResponse);
```

ResponseHandler will handle the response:
```
public T handleResponse(final HttpResponse response)
            throws HttpResponseException, IOException {
        final StatusLine statusLine = response.getStatusLine();
        final HttpEntity entity = response.getEntity();
        if (statusLine.getStatusCode() >= 300) {
            EntityUtils.consume(entity);
            throw new HttpResponseException(statusLine.getStatusCode(),
                    statusLine.getReasonPhrase());
        }
        return entity == null ? null : handleEntity(entity);
    }
```
We could also write a customized one if this is not appropriate..

it will throw error when status code is not 200. I have added a unit test case: `BatchRestClientTest.java#testInvalidClient`

Reply:
thanks for the added test